content/doc/book/system-administration/authenticating-scripted-clients.adoc

```diff
@@ -0,0 +1,378 @@
+---
+layout: section
+---
+
+ifdef::backend-html5[]
+ifndef::env-github[:imagesdir: ../../resources]
+:notitle:
+:description:
+:author:
+:email: jenkinsci-users@googlegroups.com
+:sectanchors:
+:toc: left
+endif::[]
+
+= Authenticating scripted clients
+
+To make scripted clients (such as wget) invoke operations that require
+authorization (such as scheduling a build), use HTTP BASIC
+authentication to specify the user name and the API token. 
+This is often more convenient than emulating the form-based authentication.
+
+
+Earlier versions of Jenkins require you to specify your real password,
+and it is only available when your security realm is password-based (for
+example, OpenID, Crowd and CAS plugins authenticate you without a
+password, so you simply don't have any password!) Specifying the real
+password is still supported, but it is not recommended
+because the risk of revealing password, and the human tendency to reuse
+the same password in different places.
+
+The API token is available in your personal configuration page. 
+Click your name on the top right corner on every page, 
+then click "Configure" to see your API token. 
+(The URL `+$root/me/configure+` is a good shortcut.) 
+You can also change your API token from here.
+
+Note that Jenkins does not do any authorization negotiation. 
+i.e. it immediately returns a 403 (Forbidden) response instead
+of a 401 (Unauthorized) response, so make sure to send the authentication
+information from the first request (aka "preemptive authentication").
+
+== Groovy script using cdancy/jenkins-rest
+
+The https://github.com/cdancy/jenkins-rest[cdancy/jenkins-rest client]
+greatly simplifies REST API access. 
+The following Groovy code shows how to authenticate to Jenkins and get some system info:
+
+[source,groovy]
+----
+@Grab(group='com.cdancy', module='jenkins-rest', version='0.0.18')
+
+import com.cdancy.jenkins.rest.JenkinsClient
+
+JenkinsClient client = JenkinsClient.builder()
+    .endPoint("http://127.0.0.1:8080") // Optional. Defaults to http://127.0.0.1:8080
+    .credentials("user:apiToken") // Optional.
+    .build()
+
+println(client.api().systemApi().systemInfo())
+----
+
+For additional information, see the
+https://github.com/cdancy/jenkins-rest/wiki[cdancy/jenkins-rest wiki].
+
+[[Authenticatingscriptedclients-Groovyscriptusingcommons-httpclient]]
+== Groovy script using commons-httpclient
```

Comment:
commons-httpclient was marked as end of life in 2011, let's remove this section
https://hc.apache.org/httpclient-3.x/

Reply:
Done in ca87bf15e77c277e6686d956d1e1033142b96d87